.github/workflows/vale.yml

```diff
@@ -0,0 +1,45 @@
+name: Style check
+on: [pull_request]
+
+jobs:
+  vale-lint:
+    name: runner / vale
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # OR "2" -> To retrieve the preceding commit.
+      - name: Get all changed markdown files
+        id: changed-markdown-files
+        uses: tj-actions/changed-files@v42
+        with:
+            # Avoid using single or double quotes for multiline patterns
+            files: |
+               **.md
+            separator: ","
+      - name: List all changed files markdown files
+        if: steps.changed-markdown-files.outputs.any_changed == 'true'
+        env:
+            ALL_CHANGED_FILES: ${{ steps.changed-markdown-files.outputs.all_changed_files }}
+        run: |
+            echo ${ALL_CHANGED_FILES}
+            for file in ${ALL_CHANGED_FILES}; do
+              echo "$file was changed"
+              echo "ALL_CHANGED_FILES=$ALL_CHANGED_FILES" >> $GITHUB_ENV
+            done
+      - name: No files changed?
```

Comment:
I think this step (lines 30-39) can probably be skipped: 
- Outputting that there's no changed files is somewhat redundant when there's a step earlier in the workflow to output a list of changed files. If that step is empty then there are no changed files.
- `ALL_CHANGED_FILES` can be set with `${{ 0 }}` as the initial value early in the workflow. Subsequent steps can overwrite if there are changed files that need to be tracked.